### PR TITLE
refactor!: Use multihash format for private keys

### LINKED ISSUE #4412: [suggestion] Adopt multihash format for private keys
In configurations, we use the following format for public/private keys:

```toml
public_key = "ed012014A2900349A8CB6250D8CE5399D1964766B559A37F48E8C98C1EA83F365D8A72"
private_key = { algorithm = "ed25519", payload = "67AAB998A3951A071324B815BAF425309D6AF8D2D2636A3FC2BEA01C322BC23B14A2900349A8CB6250D8CE5399D1964766B559A37F48E8C98C1EA83F365D8A72" }
```

While multihash format is used for `public_key`, why can't we use it for the private one too?

It would make config more consistent and less confusing, and will also remove necessity to set private key in the environment through 2 split variables `PRIVATE_KEY_ALGORITHM` and `PRIVATE_KEY_PAYLOAD`.

### PR DESCRIPTION
**Pull request by dima74**
## Description

Added support for encoding/decoding private keys to `multihash` module, and replaced each occurence of `PRIVATE_KEY_ALGORITHM`/`PRIVATE_KEY_PAYLOAD` pair to `PRIVATE_KEY` in multihash format

### Linked issue

Closes #4412

### Checklist

- [x] I've read `CONTRIBUTING.md`
- [x] I've used the standard signed-off commit format (or will squash just before merging)
- [ ] All applicable CI checks pass (or I promised to make them pass later)
- [ ] (optional) I've written unit tests for the code changes
- [ ] I replied to all comments after code review, marking all implemented changes with thumbs up

### CODE REVIEW

File: config/src/parameters/user.rs [hunk 1]
```diff
@@ -667,8 +601,7 @@ mod tests {
     #[test]
     fn parses_private_key_from_env() {
         let env = TestEnv::new()
-            .set("PRIVATE_KEY_ALGORITHM", "ed25519")
-            .set("PRIVATE_KEY_PAYLOAD", "8f4c15e5d664da3f13778801d23d4e89b76e94c1b94b389544168b6cb894f84f8ba62848cf767d72e7f7f4b9d2d7ba07fee33760f79abe5597a51520e292a0cb");
+            .set("PRIVATE_KEY", "8026408F4C15E5D664DA3F13778801D23D4E89B76E94C1B94B389544168B6CB894F84F8BA62848CF767D72E7F7F4B9D2D7BA07FEE33760F79ABE5597A51520E292A0CB");
```
Comment (Erigara) [inline config/src/parameters/user.rs:604]: Public key serialization looks a bit different, why?

**Review by Erigara — APPROVED** [review]

**Review by Erigara — APPROVED** [review]
LGTM, good impovment in readability of private key

File: tools/swarm/src/compose.rs [hunk 2]
```diff
@@ -325,30 +321,19 @@ struct CompactPeerEnv {
...
-            });
-
-        let (private_key_algorithm, private_key_payload) = {
-            let (algorithm, payload) = value.key_pair.private_key().clone().to_bytes();
-            (algorithm, payload)
+        let genesis_private_key = value.genesis_private_key.map(ExposedPrivateKey);
+        let genesis_file = if genesis_private_key.is_some() {
+            Some(PATH_TO_GENESIS.to_string())
+        } else {
+            None
```
Comment (0x009922) [inline tools/swarm/src/compose.rs:328]:
```suggestion
        let genesis_file = genesis_private_key.is_some().then(|| PATH_TO_GENESIS.to_string());
```

**Review by 0x009922 — APPROVED** [review]

**Comment by 0x009922** [comment]
Looking forward to merge it soon, so that I can rebase #4456 on top of it.

File: crypto/src/multihash.rs [hunk 3]
```diff
@@ -2,192 +2,170 @@
...
+mod digest_function_private {
+    #[cfg(not(feature = "std"))]
+    use alloc::string::String;
+
+    use crate::{error::ParseError, multihash::DigestFunction, Algorithm};
+
+    const ED_25519: DigestFunction = 0x1300;
+    const SECP_256_K1: DigestFunction = 0x1301;
+    const BLS12_381_G1: DigestFunction = 0x1309;
+    const BLS12_381_G2: DigestFunction = 0x130a;
```
Comment (mversic) [inline crypto/src/multihash.rs:97]: how come the constants are different for private key?

Comment (mversic) [reply crypto/src/multihash.rs:97]: ok, I see the table

**Pull request merged**